### ISSUE TITLE
updating python package to thermal nonEquilibrium 

### ISSUE THREAD

**Issue by ggange** [mutationpp/Mutationpp#179]
Hello guys,
I am trying to exploit the functionality of the MutationPP python package (got from pip). 

I wrote a small script to define a mixture using "ChemNonEqTTv" but, as soon as I tried to setState, I get this error

mix.setState(rho, T, 1)
TypeError: setState(): incompatible function arguments. The following argument types are supported:
    1. (self: mutationpp._mutationpp.Mixture, arg0: float, arg1: float, arg2: int) -> None
    2. (self: mutationpp._mutationpp.Mixture, arg0: List[float], arg1: float, arg2: int) -> None

Both rho and T are set as lists but the same happens with numpy objects...

After peaking with @BBArrosDias,  we think the problem (which I suppose is related to the input arg1 which only allows int) can be solved by modifying pyMixture.cpp adding:

     .def("setState",
           [](Mutation::Mixture &self,
                   std::vector<double> rho_i,
                   std::vector<double> T,
                   const int vars)
          {
               self.setState(rho_i.data(),T.data(),vars);
           },
           "Sets the state of the mixture using the StateModel belonging to "
           "the mixture."
           "The input variables depend on the type of StateModel being used.")

What do you think? How can I then transfer the changes to the actual python package?

Thank you!

**Comment by rubendibattista** [comment]
Ehi @ggange, can you attach the script? Such that I can use it to verify the implementation and maybe add a test.

**Comment by BBArrosDias** [comment]
Hi @rubendibattista, I think the point here is to get familiar with changing the package locally and update it.
In your opinion should we do something like this:
https://widdowquinn.github.io/coding/update-pypi-package/

I am also interested in adding more tests, and it would be nice if @ggange does this update so he can get familiar with the package if he needs to add something more in the future. 
This issue can also be a reference --  or potentially we could write a small how-to --  for future users.

**Comment by rubendibattista** [comment]
Yes. Just to be clear: you don't need to upload to PyPi the new package to have it available locally. Building the package correctly in editable mode will give you straight access to the new version. Then, once you push it upstream, we will let all the other users have access to your improvements.
